client/AI/AIClientApp.cpp

```diff
@@ -221,102 +221,103 @@ void AIClientApp::HandleMessage(const Message& msg) {
     }
 
     case Message::JOIN_GAME: {
-        if (msg.SendingPlayer() == Networking::INVALID_PLAYER_ID) {
-            if (PlayerID() == Networking::INVALID_PLAYER_ID) {
-                DebugLogger() << "AIClientApp::HandleMessage : Received JOIN_GAME acknowledgement";
-                m_networking.SetPlayerID(msg.ReceivingPlayer());
-            } else {
-                ErrorLogger() << "AIClientApp::HandleMessage : Received erroneous JOIN_GAME acknowledgement when already in a game";
+        if (PlayerID() == Networking::INVALID_PLAYER_ID) {
+            DebugLogger() << "AIClientApp::HandleMessage : Received JOIN_GAME acknowledgement";
+            const std::string& text = msg.Text();
+            try {
+                int player_id = boost::lexical_cast<int>(text);
+                m_networking.SetPlayerID(player_id);
+            } catch(...) {
+                ErrorLogger() << "AIClientApp::HandleMessage : Received erroneous JOIN_GAME acknowledgement " << text;
```

Comment:
1. I suppose it simpler for single value then serialization also it was used in HOST_ID handler.
2. OK, I'll better remove exception handler there.